src/bridge/cockpitpcpmetrics.c

```diff
@@ -675,6 +760,255 @@ cockpit_pcp_metrics_prepare (CockpitChannel *channel)
         }
     }
 
+  result = TRUE;
+
+ out:
+  g_free (instances);
+  g_free (omit_instances);
+  return result;
+}
+
+static void start_archive (CockpitPcpMetrics *self, gint64 timestamp);
+
+static gboolean
+add_archive (CockpitPcpMetrics *self,
+             const gchar *name)
+{
+  ArchiveInfo *info;
+  pmLogLabel label;
+  int rc;
+
+  info = g_new0 (ArchiveInfo, 1);
+  info->context = pmNewContext (PM_CONTEXT_ARCHIVE, name);
```

Comment:
Should be fine.  Memory will be consumed for all the .meta / .index content, but that's on the order of kilobytes.